lib/cocoapods/generator/embed_frameworks_script.rb

```diff
@@ -45,6 +45,15 @@ def script
 
           install_framework()
           {
+            if [ -e "$1" ]; then
+              version=$(defaults read "${PODS_ROOT}/Target Support Files/$(basename -s .framework "$1")/Info.plist" CFBundleShortVersionString)
```

Comment:
Hmm I apologise 1st as I did not check the new changes that have been approved in the current beta version. My bad. I took a look at #6800 #6806 . I think I had the same idea as you @dnkoutso on skipping the frameworks from being rebuilt.

Just more context on what I was trying to achieved.
@dnkoutso Trying to avoid the frameworks from rebuilding when they have already been build in the derived data. So I first checked if the framework folder already existed. 2nd, if the framework folder already existed, I compare the version of the framework with the version of the source codes (from both info.plist) to see if I need to reinstall the framework again. Avoiding unnecessary framework reinstall saves me a lot of build time when building the main project workspace. "A lot" here is just referring to cutting the build time from about a minute to a few seconds.